tests/numerics/type_tensor_test.C

```diff
@@ -60,6 +62,18 @@ private:
         CPPUNIT_ASSERT_DOUBLES_EQUAL(inverse(i,j), true_inverse(i,j), 1.e-12);
   }
 
+  void testLeftMultiply()
+  {
+    TensorValue<Real> tensor(1, 2, 0, 3, 4, 0);
+    VectorValue<Real> vector(5, 6, 0);
+    auto left_mult = vector * tensor;
+    auto right_mult = tensor * vector;
+    CPPUNIT_ASSERT_DOUBLES_EQUAL(left_mult(0), 23, 1e-12);
```

Comment:
It looks like we have several hard-coded 1e-12 constants in our unit tests. They should probably all be changed to `TOLERANCE*TOLERANCE`, but I'm OK merging as-is.